nautobot/core/views/drf_views.py

```diff
@@ -0,0 +1,836 @@
+            if "_addanother" in request.POST:
+                # If the object has clone_fields, pre-populate a new instance of the form
+                if hasattr(obj, "clone_fields"):
+                    url = f"{request.path}?{prepare_cloned_fields(obj)}"
+                    self.success_url = url
+                self.success_url = request.get_full_path()
+            else:
+                return_url = form.cleaned_data.get("return_url")
+                if return_url is not None and is_safe_url(url=return_url, allowed_hosts=request.get_host()):
+                    self.success_url = return_url
+                else:
+                    self.success_url = self.get_return_url(request, obj)
+
+    def create(self, request, *args, **kwargs):
+        """
+        request.GET: render the ObjectForm which is passed to NautobotHTMLRenderer as Response.
+        request.POST: call perform_create() which validates the form and perform the action of create.
+        Override to add more variables to Response.
+        """
+        context = {}
+        if request.method == "POST":
+            return self.perform_create(request, *args, **kwargs)
+        return Response(context)
+
+    def perform_create(self, request, *args, **kwargs):
+        """
+        Function to validate the ObjectForm and to create a new object.
+        """
+        self.obj = self.alter_obj_for_edit(self.get_object(), request, args, kwargs)
+        form_class = self.get_form_class()
+        form = form_class(data=request.POST, files=request.FILES, instance=self.obj)
+        restrict_form_fields(form, request.user)
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+    def update(self, request, *args, **kwargs):
+        """
+        request.GET: render the ObjectEditForm which is passed to NautobotHTMLRenderer as Response.
+        request.POST: call perform_update() which validates the form and perform the action of update/partial_update of an existing object.
+        Override to add more variables to Response.
+        """
+        context = {}
+        if request.method == "POST":
+            return self.perform_update(request, *args, **kwargs)
+        return Response(context)
+
+    def perform_update(self, request, *args, **kwargs):
+        """
+        Function to validate the ObjectEditForm and to update/partial_update an existing object.
+        """
+        self.obj = self.alter_obj_for_edit(self.get_object(), request, args, kwargs)
+        form_class = self.get_form_class()
+        form = form_class(data=request.POST, files=request.FILES, instance=self.obj)
+        restrict_form_fields(form, request.user)
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+
+class BulkDestroyViewMixin(NautobotViewSetMixin, bulk_mixins.BulkDestroyModelMixin):
```

Comment:
It bugs me more than it should that this isn't named `class ObjectBulkDestroyViewMixin` for consistency with the non-bulk mixins above. Same comment on `BulkUpdateViewMixin` and `BulkCreateViewMixin`.